packages/block-editor/src/components/block-has-dot/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { Fill } from '@wordpress/components';
+
+function BlockHasDot( {
+	parentReusableBlockHasEdits,
+	selectedReusableBlockHasEdits,
+} ) {
+	return (
+		<>
+			{ /* Add dot to the BlockParentSelector, if the parent block is reusable block and if the parent reusable block has edits */ }
+			<Fill name="parent-selector-has-dot">
+				{ ( { setReusableBlockHasEdits } ) =>
+					parentReusableBlockHasEdits
+						? setReusableBlockHasEdits( true )
+						: setReusableBlockHasEdits( false )
+				}
+			</Fill>
```

Comment:
Wrapping the fill in a component like this is close to what I was thinking, but it should be possible to render the `dot` directly within the fill. Something like:
```jsx
<Fill name="parent-selector-has-dot">
    <UnsavedChangesIcon />
</Fill>
```